tests/conftest.py

```diff
@@ -1509,3 +1519,14 @@ def guard_testing_venv():
             venv_was_altered = True
             error_message += f"\t* {pkg}: initial version={version_before_tests} --> after tests={version_after_tests}\n"
     assert not venv_was_altered, error_message
+
+
+@pytest.fixture(scope="function")
+def inmanta_state_dir(tmpdir_factory: "TempdirFactory") -> Iterator[str]:
+    """
+    This fixture overrides the inmanta_state_dir fixture defined in pytest-inmanta so
+    that fixtures that need to set up and write to a state directory use the same one.
+    """
+    inmanta_state_dir = tmpdir_factory.mktemp("inmanta_state_dir")
+    yield str(inmanta_state_dir)
+    inmanta_state_dir.remove(ignore_errors=True)
```

Comment:
Why ignore errors?

Reply:
After pondering for a while, I don't have a satisfying answer to this. I found very little documentation about this (https://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.remove).
I think we want to make sure that the directory is cleaned up correctly when it's no longer used but I don't know how to go about it. 
I can only think of 1 error that could occur during deletion:
- access control rights -> the current process is not allowed to remove a subdir created by another process

Reply:
True. And if that happens, something is clearly wrong. So there is no reason to hide it.